PAPER.md

```diff
@@ -0,0 +1,378 @@
+|type(4)|                                                       |
++-+-+-+-+                                                       +
+|                                                               |
++                                                               +
+|                                                               |
++                                                               +
+|                            uuid(128)                          |
++       +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|       |                                                       |
++-+-+-+-+                                                       +
+|                                                               |
++                                                               +
+|                                                               |
++                                                               +
+|                                                               |
++                                                               +
+|                                                               |
++                                                               +
+|                                                               |
++                                                               +
+|                                                               |
++                                                               +
+|                        message hash(256)                      |
++       +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|       |                                                       |
++-+-+-+-+                                                       +
+|                                                               |
++                                                               +
+|                                                               |
++                                                               +
+|                                                               |
++                                                               +
+|                                                               |
++                                                               +
+|                                                               |
++                                                               +
+|                                                               |
++                                                               +
+|                     acknowledger(256)                         |
++       +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|       |        signature len(16)      |                       |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+                       +
+|                            signature                          |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
```

Comment:
I guess the version of this messages relates implicitly to the version of the protocol previously agreed between the peers right? So it cannot happen that I'm using the protocol v1 but sending v1alpha1  messages?

Reply:
Yes. Initially, i thought about it in the same way. The discovery message with the version its emitted each `5` seconds. so the application should not send  messages to an incompatible peer. Of course, this could cause race conditions.

The problem i see clearer now is that this protocol is completely stateless (and should remain as that), as no session is intended among peers. In some way, it works like an email system, although its true that users will perceive chat semantics. Due to this reasoning, i am thinking that probably we would need to add the `version` field on every protocol message ? :thinking:

Reply:
Added version field to every message at [42e2304](https://github.com/eloylp/decentra-chat/pull/1/commits/42e230429d0c38d3ef0dbfa897d683cfad99ac3b)